emu_mps/mps_backend.py

```diff
@@ -55,7 +58,15 @@ def run(self) -> Results:
         impl = create_impl(self._sequence, self._config)
         impl.init()  # This is separate from the constructor for testing purposes.
 
-        return self._run(impl)
+        results = self._run(impl)
+        if not self._config.optimize_qubit_ordering:
+            return results
+
+        inv_perm = optimat.inv_permutation(impl.qubit_permutation)
+        permute_bitstrings(inv_perm, results)
+        permute_correlations(inv_perm, results)
```

Comment:
please also update the qubit_order in the results again to reflect the original ordering. Maybe split this logic into a new function? Something like `reorder_results`, and then you can just do
`return self.reoder_results(results)`, where it handles the lines 62 and below.

Reply:
done